crates/chain/src/persist.rs

```diff
@@ -1,33 +1,62 @@
-use core::convert::Infallible;
-
 use crate::Append;
+use alloc::boxed::Box;
+use core::fmt;
+
+#[derive(Debug)]
+/// Errors returned by [`PersistBackend`] when writing or loading changes.
+pub enum PersistBackendError {
+    /// Happens when there is an standard IO error writing to the backend file.
+    IoError,
```

Comment:
Shouldn't we include the `std::io::Error` here? The variant will need to be `#[cfg(feature = "std")]` of course.

Reply:
Yes we should. I was unaware you are able to feature gate variants like this. The marco just goes above the variant?